tests/e2e/e2e.go

```diff
@@ -51,39 +54,39 @@ var _ = ginkgo.Describe("[HyperSDK APIs]", func() {
 	tc := e2e.NewTestContext()
 	require := require.New(tc)
 
-	ginkgo.It("Ping", func() {
-		expectedBlockchainID := e2e.GetEnv(tc).GetNetwork().GetSubnet(vmName).Chains[0].ChainID
+	ginkgo.It("Ping", asyncExecution(func() {
+		expectedBlockchainID := e2e.GetEnv(tc).GetNetwork().GetSubnet(networkConfig.Name()).Chains[0].ChainID
 		workload.Ping(tc.DefaultContext(), require, getE2EURIs(tc, expectedBlockchainID))
-	})
+	}))
 
-	ginkgo.It("StableNetworkIdentity", func() {
+	ginkgo.It("StableNetworkIdentity", asyncExecution(func() {
 		hardcodedHostPort := "http://localhost:9650"
-		fixedNodeURL := hardcodedHostPort + "/ext/bc/" + vmName
+		fixedNodeURL := hardcodedHostPort + "/ext/bc/" + networkConfig.Name()
 
 		c := jsonrpc.NewJSONRPCClient(fixedNodeURL)
 		_, _, chainIDFromRPC, err := c.Network(tc.DefaultContext())
 		require.NoError(err)
-		expectedBlockchainID := e2e.GetEnv(tc).GetNetwork().GetSubnet(vmName).Chains[0].ChainID
+		expectedBlockchainID := e2e.GetEnv(tc).GetNetwork().GetSubnet(networkConfig.Name()).Chains[0].ChainID
 		require.Equal(expectedBlockchainID, chainIDFromRPC)
-	})
+	}))
 
-	ginkgo.It("GetNetwork", func() {
-		expectedBlockchainID := e2e.GetEnv(tc).GetNetwork().GetSubnet(vmName).Chains[0].ChainID
+	ginkgo.It("GetNetwork", asyncExecution(func() {
```

Comment:
Rather than use `asyncExecution`, could we make sure that the tests do not interfere with each other? I assume this is because we are using the same spending key for all of the tests. The simplest solution to this with this change is to create an isolated spending key for the newly added test registry. The more functionally complete version (good for a separate PR), would be to have each registered test describe the number of keys/amount of funds it requires to execute and then handle distributing funds to it in the background.

Reply:
ok, found that ginkgo has the `Serial` flag that would do the same thing for us.. used that instead.